test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -191,49 +189,60 @@ private static void doTest()
 
         robot.delay(DELAY);
         if (componentShown || componentHidden) {
-            System.err.print("Events triggered are: ");
-            for (int j = 0; j < events.size();
-                System.err.print(events.get(j) + "; "), j++);
-            System.err.println();
-            throw new RuntimeException(
-                "FAIL: ComponentEvent triggered when frame is set to normal state");
+            printErrEvents("ComponentEvent triggered when frame set to normal");
         }
     }
 
-    private static void doTestForComponent(int i)
+    private static void printErrEvents(String errorMsg) {
+        System.err.print("Events triggered are: ");
+        for (int j = 0; j < events.size(); j++) {
+            System.err.print(events.get(j) + "; ");
+        }
```

Comment:
```suggestion
        for (ComponentEvent event : events) {
            System.err.print(event + "; ");
        }
```
You don't use the index for anything else but getting an element, you can use the improved `for`-loop.

If you do, you can replace the type of `events` from `java.util.List` to `java.util.Collection`.

*Strictly*, you should iterate over `events` while holding the lock to the `events` object, even though there's no strong need for it since the list of events can't be modified.

```java
        synchronized (events) {
            for (ComponentEvent event : events) {
                System.err.print(event + "; ");
            }
        }
```